util/reggen/gen_cheader.py

```diff
@@ -200,7 +207,10 @@ def gen_cdefines_module_param(outstr: TextIO,
 
     if param.desc is not None:
         genout(outstr, format_comment(first_line(param.desc)))
-    define_name = as_define(module_name + '_PARAM_' + param.name)
+    # Heuristic: if the name already has underscores, it's already snake_case,
+    # otherwise, assume StudlyCaps and covert it to snake_case.
+    param_name = param.name if '_' in param.name else to_snake_case(param.name)
```

Comment:
maybe instead of having this check here, we can expressly enforce through the reggen tool (different PR) that the hardware guys always encode the param in camel case (this is really what we do anyways).

That way you can just directly push it to snake case. 

Also, should we consider unifying with the code [here](https://github.com/lowRISC/opentitan/blob/master/util/topgen/c.py)?

otherwise, this LGTM.

Reply:
@cfrantz just pointed out he needed this check because `pwrmgr`, `alert_handler` and `rv_plic` have parameters not in camel case. 

let me go have a look at this. i know our style guide in general encourages parameters to be in camel case..but we do allow constants to be in snake case.

Reply:
It might also be worth looking at `_to_pascal_case` in `hw/ip/lc_ctrl/rtl/lc_ctrl_state_pkg.sv.tpl`. I think @msfschaffner solved basically the same problem there: maybe we could share the code?

Reply:
`to_pascal_case` is the converse of the operation I'm performing here.

Your point still stands: there should be a single implementation of case-conversion functions in the python tooling and all users needing to do case conversions should use the common functions (https://github.com/lowRISC/opentitan/issues/6310).